### ISSUE TITLE
setup module: ValueError: No JSON object could be decoded and No module named module_exec.setup

### ISSUE THREAD

**Issue by infernix** [ansible/ansible#16152]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel d21cd16a91) last updated 2016/06/05 17:24:08 (GMT +200)
  lib/ansible/modules/core: (detached HEAD eb1cb94380) last updated 2016/06/06 14:54:18 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD fa65f4dc2b) last updated 2016/06/06 14:54:18 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

Also happens with 2.1.0 stable.
##### CONFIGURATION

No changes (running straight from source dir)
##### OS / ENVIRONMENT

Ansible running host is Debian sid, ansible target host is Debian Jessie.
##### SUMMARY

The setup module fails to run in both 2.1.0 stable and devel d21cd16a91 with the error `ValueError: No JSON object could be decoded` which breaks all executions against this host.
##### STEPS TO REPRODUCE

```
ansible -vvvv -u root -m setup -i testinventory ansible-target-host
```

##### EXPECTED RESULTS

Setup should run without issue
##### ACTUAL RESULTS

I have bisected this and uncovered two problems. The issue in current devel HEAD is:

```
Loaded callback minimal of type stdout, v2.0
<ansible-target-host> ESTABLISH SSH CONNECTION FOR USER: root
<ansible-target-host> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/infernix/.ansible/cp/ansible-ssh-%h-%p-%r ansible-target-host '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1465217692.09-264707081140791 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1465217692.09-264707081140791 `" )'"'"''
<ansible-target-host> PUT /tmp/tmpDOSaTv TO /root/.ansible/tmp/ansible-tmp-1465217692.09-264707081140791/setup
<ansible-target-host> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/infernix/.ansible/cp/ansible-ssh-%h-%p-%r '[ansible-target-host]'
<ansible-target-host> ESTABLISH SSH CONNECTION FOR USER: root
<ansible-target-host> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/infernix/.ansible/cp/ansible-ssh-%h-%p-%r -tt ansible-target-host '/bin/sh -c '"'"'LANG=en_GB.UTF-8 LC_ALL=en_GB.UTF-8 LC_MESSAGES=en_GB.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1465217692.09-264707081140791/setup'"'"''
ansible-target-host | FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "setup"
    }, 
    "module_stderr": "OpenSSH_7.2p2 Debian-2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /home/infernix/.ssh/config\r\ndebug1: /home/infernix/.ssh/config line 27: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 18985\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 1\r\nShared connection to ansible-target-host closed.\r\n", 
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_56STFM/ansible_module_setup.py\", line 127, in <module>\r\n    main()\r\n  File \"/tmp/ansible_56STFM/ansible_module_setup.py\", line 117, in main\r\n    supports_check_mode = True,\r\n  File \"/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py\", line 533, in __init__\r\n    self.params = self._load_params()\r\n  File \"/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py\", line 1351, in _load_params\r\n    params = json_dict_unicode_to_bytes(json.loads(MODULE_COMPLEX_ARGS))\r\n  File \"/usr/lib/python2.7/json/__init__.py\", line 338, in loads\r\n    return _default_decoder.decode(s)\r\n  File \"/usr/lib/python2.7/json/decoder.py\", line 366, in decode\r\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\r\n  File \"/usr/lib/python2.7/json/decoder.py\", line 384, in raw_decode\r\n    raise ValueError(\"No JSON object could be decoded\")\r\nValueError: No JSON object could be decoded\r\n", 
    "msg": "MODULE FAILURE", 
    "parsed": false
}
```

I git bisected that to 452034564c32be415f1d2289339bf7bbbf23d273

Furthermore, while doing git bisect, I also had this:

```
Loaded callback minimal of type stdout, v2.0
<ansible-target-host> ESTABLISH SSH CONNECTION FOR USER: root
<ansible-target-host> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/infernix/.ansible/cp/ansible-ssh-%h-%p-%r ansible-target-host '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1465217659.85-21836127905024 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1465217659.85-21836127905024 `" )'"'"''
<ansible-target-host> PUT /tmp/tmpetYjmS TO /root/.ansible/tmp/ansible-tmp-1465217659.85-21836127905024/setup
<ansible-target-host> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/infernix/.ansible/cp/ansible-ssh-%h-%p-%r '[ansible-target-host]'
<ansible-target-host> ESTABLISH SSH CONNECTION FOR USER: root
<ansible-target-host> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/infernix/.ansible/cp/ansible-ssh-%h-%p-%r -tt ansible-target-host '/bin/sh -c '"'"'LANG=en_GB.UTF-8 LC_ALL=en_GB.UTF-8 LC_MESSAGES=en_GB.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1465217659.85-21836127905024/setup'"'"''
ansible-target-host | FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "setup"
    }, 
    "module_stderr": "OpenSSH_7.2p2 Debian-2, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /home/infernix/.ssh/config\r\ndebug1: /home/infernix/.ssh/config line 27: Applying options for *\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 18985\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 1\r\nShared connection to ansible-target-host closed.\r\n", 
    "module_stdout": "/usr/bin/python: No module named module_exec.setup\r\n", 
    "msg": "MODULE FAILURE", 
    "parsed": false
}
```

I git bisected that to 647123ee187f288d6c152f328a7dd7ff59f2d754

In both cases. logging in to the target host and doing an `explode` and `execute` on the copied module worked fine, but when i then removed the exploded `ansible` dir, it returned the error again. Example for the first problem:

```
root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465218320.95-8271008413542# ls -ltra
total 88
drwxr-xr-x 3 root root    52 Jun  6 15:06 ..
drwx------ 2 root root    21 Jun  6 15:06 .
-rwx------ 1 root root 88550 Jun  6 15:06 setup.py

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465218320.95-8271008413542# python setup.py 
Traceback (most recent call last):
  File "/tmp/ansible_HlkycH/ansible_module_setup.py", line 127, in <module>
    main()
  File "/tmp/ansible_HlkycH/ansible_module_setup.py", line 117, in main
    supports_check_mode = True,
  File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 533, in __init__
    self.params = self._load_params()
  File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 1351, in _load_params
    params = json_dict_unicode_to_bytes(json.loads(MODULE_COMPLEX_ARGS))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465218320.95-8271008413542# python setup.py explode
Module expanded into:
/root/.ansible/tmp/ansible-tmp-1465218320.95-8271008413542/debug_dir

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465218320.95-8271008413542# python setup.py execute

{"invocation": ... }

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465218320.95-8271008413542# python setup.py 
Traceback (most recent call last):
  File "/tmp/ansible_Zb68oQ/ansible_module_setup.py", line 127, in <module>
    main()
  File "/tmp/ansible_Zb68oQ/ansible_module_setup.py", line 117, in main
    supports_check_mode = True,
  File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 533, in __init__
    self.params = self._load_params()
  File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 1351, in _load_params
    params = json_dict_unicode_to_bytes(json.loads(MODULE_COMPLEX_ARGS))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

Note that there is a preceding newline before the JSON output. I did dump the entire JSON output to file and verified it with jsonlint-php; it returned no errors.

And the second:

```
root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# ls -la
total 76
drwx------  2 root root    18 Jun  6 14:45 .
drwxr-xr-x 10 root root  4096 Jun  6 14:32 ..
-rw-------  1 root root 69883 Jun  6 14:32 setup

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# python setup 
/usr/bin/python: No module named module_exec.setup

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# python setup explode
Module expanded into:
ansible

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# python setup execute
{"invocation": ... }

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# python setup
{"invocation": ... }

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# rm -rf ansible/

root@ansible-target-host:~/.ansible/tmp/ansible-tmp-1465216325.61-21939240646370# python setup
/usr/bin/python: No module named module_exec.setup
```

**Comment by alikins** [comment]
(reformatting of stderr output from initial report for readability)
module_stderr

```
OpenSSH_7.2p2 Debian-2, OpenSSL 1.0.2g  1 Mar 2016
debug1: Reading configuration data /home/infernix/.ssh/config
debug1: /home/infernix/.ssh/config line 27: Applying options for *
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: auto-mux: Trying existing master
debug2: fd 3 setting O_NONBLOCK
debug2: mux_client_hello_exchange: master version 4
debug3: mux_client_forwards: request forwardings: 0 local, 0 remote
debug3: mux_client_request_session: entering
debug3: mux_client_request_alive: entering
debug3: mux_client_request_alive: done pid = 18985
debug3: mux_client_request_session: session request sent
debug1: mux_client_request_session: master session id: 2
debug3: mux_client_read_packet: read header failed: Broken pipe
debug2: Received exit status from master 1
Shared connection to ansible-target-host closed.
```

module_stdout

```
"module_stdout":
  Traceback (most recent call last):
  File "/tmp/ansible_56STFM/ansible_module_setup.py", line 127, in <module>
    main()
  File "/tmp/ansible_56STFM/ansible_module_setup.py", line 117, in main
    supports_check_mode = True,
  File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 533, in __init__
    self.params = self._load_params()
  File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 1351, in _load_params
    params = json_dict_unicode_to_bytes(json.loads(MODULE_COMPLEX_ARGS))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by abadger** [comment]
```
 File "/usr/lib/pymodules/python2.7/ansible/module_utils/basic.py", line 533, in __init__
```

It looks like you have an older version of ansible installed on the target and the module is picking up that installation instead of the one contained in the ansiballz zip file.  We have code that's supposed to prevent that.  Can you tell me anything about how /usr/lib/pymodules/python2.7 is being added to the path?

**Comment by infernix** [comment]
Does this help?

```
root@spirit:~# dpkg -S /usr/lib/pymodules/python2.7/ansible/module_utils/basic.py
dpkg-query: no path found matching pattern /usr/lib/pymodules/python2.7/ansible/module_utils/basic.py

root@spirit:~# ls -ltra /usr/lib/pymodules/python2.7/ansible/module_utils/basic.py
lrwxrwxrwx 1 root root 49 Mar 24 14:24 /usr/lib/pymodules/python2.7/ansible/module_utils/basic.py -> /usr/share/pyshared/ansible/module_utils/basic.py

root@spirit:~# dpkg -S /usr/share/pyshared/ansible/module_utils/basic.py
ansible: /usr/share/pyshared/ansible/module_utils/basic.py

root@spirit:~# dpkg -s ansible
Package: ansible
Status: install ok installed
Priority: optional
Section: admin
Installed-Size: 6877
Maintainer: Ansible, Inc. <support@ansible.com>
Architecture: all
Version: 2.0.1.0-0.git201603240000..~unstable
Depends: python, python-support (>= 0.90), python-jinja2, python-yaml, python-paramiko, python-httplib2, python-six, python-crypto (>= 2.6), python-setuptools, sshpass
Conffiles:
 /etc/ansible/ansible.cfg 248f57c4cb0f11fe8fd5f39b12ac8205
 /etc/ansible/hosts 1564b951dc7c8511c6f9ee842653c541
Description: A radically simple IT automation platform
 A radically simple IT automation platform that makes your applications and
 systems easier to deploy. Avoid writing scripts or custom code to deploy and
 update your applications— automate in a language that approaches plain English,
 using SSH, with no agents to install on remote systems.
Homepage: http://ansible.github.com/

root@spirit:~# python -v
# installing zipimport hook
import zipimport # builtin
# installed zipimport hook
# /usr/lib/python2.7/site.pyc matches /usr/lib/python2.7/site.py
import site # precompiled from /usr/lib/python2.7/site.pyc
# /usr/lib/python2.7/os.pyc matches /usr/lib/python2.7/os.py
import os # precompiled from /usr/lib/python2.7/os.pyc
import errno # builtin
import posix # builtin
# /usr/lib/python2.7/posixpath.pyc matches /usr/lib/python2.7/posixpath.py
import posixpath # precompiled from /usr/lib/python2.7/posixpath.pyc
# /usr/lib/python2.7/stat.pyc matches /usr/lib/python2.7/stat.py
import stat # precompiled from /usr/lib/python2.7/stat.pyc
# /usr/lib/python2.7/genericpath.pyc matches /usr/lib/python2.7/genericpath.py
import genericpath # precompiled from /usr/lib/python2.7/genericpath.pyc
# /usr/lib/python2.7/warnings.pyc matches /usr/lib/python2.7/warnings.py
import warnings # precompiled from /usr/lib/python2.7/warnings.pyc
# /usr/lib/python2.7/linecache.pyc matches /usr/lib/python2.7/linecache.py
import linecache # precompiled from /usr/lib/python2.7/linecache.pyc
# /usr/lib/python2.7/types.pyc matches /usr/lib/python2.7/types.py
import types # precompiled from /usr/lib/python2.7/types.pyc
# /usr/lib/python2.7/UserDict.pyc matches /usr/lib/python2.7/UserDict.py
import UserDict # precompiled from /usr/lib/python2.7/UserDict.pyc
# /usr/lib/python2.7/_abcoll.pyc matches /usr/lib/python2.7/_abcoll.py
import _abcoll # precompiled from /usr/lib/python2.7/_abcoll.pyc
# /usr/lib/python2.7/abc.pyc matches /usr/lib/python2.7/abc.py
import abc # precompiled from /usr/lib/python2.7/abc.pyc
# /usr/lib/python2.7/_weakrefset.pyc matches /usr/lib/python2.7/_weakrefset.py
import _weakrefset # precompiled from /usr/lib/python2.7/_weakrefset.pyc
import _weakref # builtin
# /usr/lib/python2.7/copy_reg.pyc matches /usr/lib/python2.7/copy_reg.py
import copy_reg # precompiled from /usr/lib/python2.7/copy_reg.pyc
# /usr/lib/python2.7/traceback.pyc matches /usr/lib/python2.7/traceback.py
import traceback # precompiled from /usr/lib/python2.7/traceback.pyc
# /usr/lib/python2.7/sysconfig.pyc matches /usr/lib/python2.7/sysconfig.py
import sysconfig # precompiled from /usr/lib/python2.7/sysconfig.pyc
# /usr/lib/python2.7/re.pyc matches /usr/lib/python2.7/re.py
import re # precompiled from /usr/lib/python2.7/re.pyc
# /usr/lib/python2.7/sre_compile.pyc matches /usr/lib/python2.7/sre_compile.py
import sre_compile # precompiled from /usr/lib/python2.7/sre_compile.pyc
import _sre # builtin
# /usr/lib/python2.7/sre_parse.pyc matches /usr/lib/python2.7/sre_parse.py
import sre_parse # precompiled from /usr/lib/python2.7/sre_parse.pyc
# /usr/lib/python2.7/sre_constants.pyc matches /usr/lib/python2.7/sre_constants.py
import sre_constants # precompiled from /usr/lib/python2.7/sre_constants.pyc
import _locale # builtin
# /usr/lib/python2.7/_sysconfigdata.pyc matches /usr/lib/python2.7/_sysconfigdata.py
import _sysconfigdata # precompiled from /usr/lib/python2.7/_sysconfigdata.pyc
# /usr/lib/python2.7/plat-x86_64-linux-gnu/_sysconfigdata_nd.pyc matches /usr/lib/python2.7/plat-x86_64-linux-gnu/_sysconfigdata_nd.py
import _sysconfigdata_nd # precompiled from /usr/lib/python2.7/plat-x86_64-linux-gnu/_sysconfigdata_nd.pyc
# /usr/lib/python2.7/sitecustomize.pyc matches /usr/lib/python2.7/sitecustomize.py
import sitecustomize # precompiled from /usr/lib/python2.7/sitecustomize.pyc
import encodings # directory /usr/lib/python2.7/encodings
# /usr/lib/python2.7/encodings/__init__.pyc matches /usr/lib/python2.7/encodings/__init__.py
import encodings # precompiled from /usr/lib/python2.7/encodings/__init__.pyc
# /usr/lib/python2.7/codecs.pyc matches /usr/lib/python2.7/codecs.py
import codecs # precompiled from /usr/lib/python2.7/codecs.pyc
import _codecs # builtin
# /usr/lib/python2.7/encodings/aliases.pyc matches /usr/lib/python2.7/encodings/aliases.py
import encodings.aliases # precompiled from /usr/lib/python2.7/encodings/aliases.pyc
# /usr/lib/python2.7/encodings/utf_8.pyc matches /usr/lib/python2.7/encodings/utf_8.py
import encodings.utf_8 # precompiled from /usr/lib/python2.7/encodings/utf_8.pyc
Python 2.7.9 (default, Mar  1 2015, 12:57:24) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
dlopen("/usr/lib/python2.7/lib-dynload/readline.x86_64-linux-gnu.so", 2);
import readline # dynamically loaded from /usr/lib/python2.7/lib-dynload/readline.x86_64-linux-gnu.so
>>> 

```

**Comment by infernix** [comment]
Found it!

`/usr/local/lib/python2.7/dist-packages/` contained `easy-install.pth`:

``` python
import sys; sys.__plen = len(sys.path)
./plugwiselib-0.2dev-py2.7.egg
./crcmod-1.7-py2.7-linux-x86_64.egg
./plugwise2py-2.0dev-py2.7.egg
/usr/lib/pymodules/python2.7
./distribute-0.7.3-py2.7.egg
./pip-7.1.2-py2.7.egg
import sys; new=sys.path[sys.__plen:]; del sys.path[sys.__plen:]; p=getattr(sys,'__egginsert',0); sys.path[p:p]=new; sys.__egginsert = p+len(new)
```

Bad easy-install. Very bad.

Removed the include to /usr/lib/pymodules/python2.7 and problem went away. Sorry for the noise!

**Comment by abadger** [comment]
EDIT:  Thanks!  easy-install.pth may be the clue we need!

I've just been working on this ticket: https://github.com/ansible/ansible/issues/16187   It _may_ be pertinent to this problem but it's hard to tell as the symptoms are mostly different. (My theory is that the symptoms are different because of different versions of ansible installed on the target machine).

Try the pull request that fixes that issue: https://github.com/ansible/ansible/pull/16238  and see if it solves this problem as well.

**Comment by abadger** [comment]
Sorry about the earlier comment -- github apparently only updates the thread in my browser some of the time.  I didn't have your comment on easy-install.pth yet.  the easy-install clue makes it look even more likely that  #16238 will fix this problem.  So hopefully with that change applied things will work even if the easy-install.pth is present.

**Comment by infernix** [comment]
NP. PR #16238 fixes it for me (commit 0bc6fa337c3a43327cb55e895b52eca2214bf202) :+1: 

Thanks!

**Comment by abadger** [comment]
Thanks for testing!  PR merged to devel and stable-2.1.  Should be in our next 2.1.x release.
